### PR TITLE
GEN-799 | Extract `useGetCartEntry` helper hook

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Extract helper hook to parse redeemed camapign to internal cart campaign type

## Justify why they are needed

- Nice to not spread out the parsing of this type too much (single source of truth)

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

File: apps/store/src/pages/checkout/index.tsx [hunk 1]
```diff
@@ -50,18 +47,8 @@ const NextCheckoutPage: NextPage<NextPageProps> = (props) => {
...
-              id: campaign.id,
-              code: campaign.code,
-              discountExplanation: getDiscountExplanation(campaign.discount),
-              discountDurationExplanation: getDiscountDurationExplanation(
-                campaign.discount,
-                shopSession.cart.cost.gross,
-              ),
-            },
-          ]
+      list: shopSession.cart.redeemedCampaign
```
Comment (alebedev) [inline apps/store/src/pages/checkout/index.tsx:50]: Why keep `campaigns.list` ?

**Review by alebedev — APPROVED** [review]

File: apps/store/src/components/CartInventory/CartInventory.helpers.tsx [hunk 2]
```diff
@@ -79,3 +82,22 @@ const getTierLevelDisplayName = (item: ShopSession['cart']['entries'][number]) =
     ? item.variant.displayName
     : undefined
 }
+
+type GetCartCampaignFunction = (total: Money, campaign: RedeemedCampaign) => CartCampaign
+
+export const useGetCartCampaign = (): GetCartCampaignFunction => {
+  const getDiscountExplanation = useGetDiscountExplanation()
+  const getDiscountDurationExplanation = useGetDiscountDurationExplanation()
```
Comment (alebedev) [inline apps/store/src/components/CartInventory/CartInventory.helpers.tsx:90]: Are we always using hooks above together?  Maybe it should be one hook?

File: apps/store/src/pages/checkout/index.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/pages/checkout/index.tsx:50]: I will clean that up in a future PR

File: apps/store/src/components/CartInventory/CartInventory.helpers.tsx (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/components/CartInventory/CartInventory.helpers.tsx:90]: True, but at least one of them is a utility used elsewhere

But I will double check

**Comment by robinandeer** [comment]
@robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2817).

**Comment by robinandeer** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2817) rebased this pull request as part of a merge.

**Comment by robinandeer** [comment]
@robinandeer merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2817).

**Pull request merged**